ingest.py

```diff
@@ -139,6 +139,86 @@ def does_vectorstore_exist(persist_directory: str) -> bool:
                 return True
     return False
 
+
+def prompt_user():
+    """
+    This function prompts the user to select an existing directory or create a new one to store source material.
+    If an existing directory is selected, it checks if the directory is empty and prompts the user to create files
+    in the directory if it is empty. It sets the directory paths as environment variables and returns them.
+
+    Returns:
+    selected_directory_path (str): The path of the selected directory.
+    selected_db_path (str): The path of the database directory for the selected directory.
+    """
+
+    def _display_directories():
+        """
+        This function displays the list of existing directories in the ./sources directory.
+        """
+        print("Existing directories in ./sources:")
+        directories = os.listdir("./sources")
```

Comment:
This will list those annoying folders generated by Mac for example: `.DS_Store`, or if you have `.gitkeep` or similar files inside it will also list them

Reply:
Thanks for catching that possibility, I've also added logic to only list directories.
Tested against the source directory containing files, dotfiles and dotfolders.